### PR TITLE
Dilpreet | Stage | Inventory Adjustment Prod Bugs

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 1]
```diff
@@ -1392,15 +1393,19 @@ export class LedgerComponent implements OnInit, OnDestroy {
     }
 
     public showUpdateLedgerModal(txn: ITransactionItem, type: string) {
-        let transactions: TransactionsResponse = null;
-        this.store.pipe(select(t => t?.ledger?.transactionsResponse), take(1)).subscribe(trx => transactions = trx);
-        if (transactions) {
-            this.store.dispatch(this.ledgerActions.setAccountForEdit(this.lc.accountUnq));
+        if (txn?.adjustmentEntry) {
+            this.router.navigate([`/pages/inventory/v2/product/adjust/${txn?.description}`]);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:1397]: Check requirement basis

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 2]
```diff
@@ -156,6 +158,16 @@ export class AdjustInventoryListComponent implements OnInit, OnDestroy {
             }
         });
 
+        this.componentStore.organistationMode$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.isCompany = this.generalService.currentOrganizationType !== OrganizationType.Branch && response?.length > 1;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts:163]:
```suggestion
                this.isCompany = this.generalService.currentOrganizationType !== OrganizationType.Branch && response?.length > 1;
                if (!this.isCompany) {
            this.displayedColumns.push('action');
        }
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 3]
```diff
@@ -156,6 +158,16 @@ export class AdjustInventoryListComponent implements OnInit, OnDestroy {
...
+            }
+        });
+
+        if (!this.isCompany) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts]:
```suggestion

```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 4]
```diff
@@ -156,6 +158,16 @@ export class AdjustInventoryListComponent implements OnInit, OnDestroy {
...
+            this.displayedColumns.push('action');
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts]:
```suggestion
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 5]
```diff
@@ -156,6 +158,16 @@ export class AdjustInventoryListComponent implements OnInit, OnDestroy {
...
+        }
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts]:
```suggestion
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/utility/adjust-inventory-list.store.ts [hunk 6]
```diff
@@ -35,6 +35,7 @@ export class AdjustInventoryListComponentStore extends ComponentStore<AdjustInve
 
     public activeCompany$: Observable<any> = this.select(this.store.select(state => state.session.activeCompany), (response) => response);
     public universalDate$: Observable<any> = this.select(this.store.select(state => state.session.applicationDate), (response) => response);
+    public organistationMode$: Observable<any> = this.select(this.store.select(state => state.settings.branches), (response) => response);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/utility/adjust-inventory-list.store.ts]:
```suggestion
    public organisationMode$: Observable<any> = this.select(this.store.select(state => state.settings.branches), (response) => response);
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts [hunk 7]
```diff
@@ -156,6 +158,16 @@ export class AdjustInventoryListComponent implements OnInit, OnDestroy {
             }
         });
 
+        this.componentStore.organistationMode$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory-list/adjust-inventory-list.component.ts]:
```suggestion
        this.componentStore. organisationMode$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.html [hunk 8]
```diff
@@ -34,6 +34,7 @@
                     </div>
                     <div class="col-xxl-3 mt-3 col-xl-3 col-lg-6 col-md-6">
                         <input-field
+                            [readonly]="referenceNumber"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.html]: Use boolean value or this

```suggestion
                            [readonly]="!!referenceNumber"
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.html [hunk 9]
```diff
@@ -305,22 +306,44 @@
...
-                                                "
-                                                [aria-label]="checkboxLabel(element)"
+                                                (change)="$event ? selection.toggle(element) : null"
+                                                [checked]="selection.isSelected(element)"
                                                 [disabled]="isEntityStockGroup"
                                             >
                                                 {{ element.variant?.name }}
                                             </mat-checkbox>
+                                            <mat-checkbox
+                                                *ngIf="isEntityStockGroup && !updateMode"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.html]:
```suggestion
                                                *ngIf="isEntityStockGroup"
```

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.html [hunk 10]
```diff
@@ -305,22 +306,44 @@
...
+                                                *ngIf="updateMode && !isEntityStockGroup"
+                                                color="primary"
+                                                class="mr-2"
+                                                (click)="$event.stopPropagation()"
+                                                (change)="$event ? selection.toggle(element) : null"
+                                                [checked]="selection.isSelected(element)"
+                                            >
+                                                {{ element.variant?.name }}
+                                            </mat-checkbox>
+                                            <mat-checkbox
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.html]: remove this unused check box

File: apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.ts [hunk 11]
```diff
@@ -543,6 +551,7 @@ export class AdjustInventoryComponent implements OnInit {
         this.stockReportRequest.count = API_COUNT_LIMIT;
         this.stockReportRequest.inventoryType = this.inventoryType?.toUpperCase();
         this.stockReportRequest.branchUniqueNames = this.generalService.currentBranchUniqueName ? [this.generalService.currentBranchUniqueName] : [];
+        this.stockReportRequest.warehouseUniqueNames = this.adjustInventoryCreateEditForm.get('warehouseUniqueName')?.value ? [this.adjustInventoryCreateEditForm.get('warehouseUniqueName')?.value] : [];
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/adjust-inventory/adjust-inventory.component.ts]: create const variable for "this.adjustInventoryCreateEditForm.get('warehouseUniqueName')?.value"

and use below line

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 1 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/ledger/ledger.component.ts:1397]: As confirmed with Rishi and Aashish this key is correct. if any question then talks to them.

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**